samtranslator/model/eventsources/push.py

```diff
@@ -96,6 +96,7 @@ class Schedule(PushEventSource):
         "Schedule": PropertyType(True, is_str()),
         "Input": PropertyType(False, is_str()),
         "Enabled": PropertyType(False, is_type(bool)),
+        "State": PropertyType(False, is_str()),
```

Comment:
Why do we want to add the `State` property? 

Essentially we just want to add intrinsic support (at least for `Fn::If` and `!Ref`). So, what we actually should do is:
- If `Enabled` is an intrinsic, just assign it to `events_rule.State`
- If `Enabled` is a string, keep it as how we handle it today (for sake of not introducing breaking change)

Reply:
okay, looking at this [comment](https://github.com/aws/serverless-application-model/issues/1329#issuecomment-1157759679), and realizing adding intrinsic support to `Enabled` also introduces breaking change, I think adding `State` property makes sense to me now.